tests/test_htmIndex.py

```diff
@@ -165,47 +174,159 @@ def testAgainstPersisted(self):
         self.assertTrue(self.test_butler.datasetExists('ref_cat', data_id))
         ref_cat = self.test_butler.get('ref_cat', data_id)
         if REGENERATE_COMPARISON:
-            os.unlink(self.test_cat_path)
+            if os.path.exists(self.test_cat_path):
+                os.unlink(self.test_cat_path)
             ref_cat.writeFits(self.test_cat_path)
             self.fail("New comparison data written; unset REGENERATE_COMPARISON in order to proceed")
 
         ex1 = ref_cat.extract('*')
-        ex2 = self.test_cat.extract('*')
+        test_cat = afwTable.SimpleCatalog.readFits(self.test_cat_path)
+
+        ex2 = test_cat.extract('*')
         self.assertEqual(set(ex1.keys()), set(ex2.keys()))
         for kk in ex1:
             np.testing.assert_array_equal(ex1[kk], ex2[kk])
 
+    @staticmethod
+    def makeConfig(withMagErr=False, withRaDecErr=False, withPm=False, withPmErr=False,
+                   withParallax=False, withParallaxErr=False):
+        """Make a config for IngestIndexedReferenceTask
+
+        This is primarily intended to simplify tests of config validation,
+        so fields that are not validated are not set.
+        However, it can calso be used to reduce boilerplate in other tests.
+        """
+        config = IngestIndexedReferenceTask.ConfigClass()
+        config.ra_name = 'ra_icrs'
+        config.dec_name = 'dec_icrs'
+        config.mag_column_list = ['a', 'b']
+
+        if withMagErr:
+            config.mag_err_column_map = {'a': 'a_err', 'b': 'b_err'}
+
+        if withRaDecErr:
+            config.ra_err_name = "ra_err"
+            config.dec_err_name = "dec_err"
+
+        if withPm:
+            config.pm_ra_name = "pm_ra"
+            config.pm_dec_name = "pm_dec"
+
+        if withPmErr:
+            config.pm_ra_err_name = "pm_ra_err"
+            config.pm_dec_err_name = "pm_dec_err"
+
+        if withParallax:
+            config.parallax_name = "parallax"
+
+        if withParallaxErr:
+            config.parallax_err_name = "parallax_err"
+
+        if withPm or withParallax:
+            config.epoch_name = "unixtime"
+            config.epoch_format = "unix"
+            config.epoch_scale = "utc"
+
+        return config
+
+    def testValidateRaDecMag(self):
+        config = self.makeConfig()
+        config.validate()
+
+        for name in ("ra_name", "dec_name", "mag_column_list"):
+            with self.subTest(name=name):
+                config = self.makeConfig()
+                setattr(config, name, None)
+                with self.assertRaises(ValueError):
```

Comment:
I'm afraid I just don't understand the test here.  Why would it raise?

Reply:
The three config items being tested all must have non-None values, so if any value is None then validation should fail.